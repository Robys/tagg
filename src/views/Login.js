import {useState} from 'react'
import {Redirect} from 'react-router-dom'
import {Paper,Button,CardHeader} from '@material-ui/core'
import MuiAlert from '@material-ui/lab/Alert';
import {LOGIN} from '../api'
import {Footer} from '../utils/SiteContent'

function Alert(props) {
    return <MuiAlert elevation={6} variant="filled" {...props} />;
  }

/** Login screen provide the access to the dashboard (main screen of the application)
 * @constructor
 * @param {string} email - users email
 * @param {string} password - users password
 */

function Login (props){

    const [email,SetEmail] = useState()
    const [password,SetPassword] = useState()

    const [error,SetError] = useState({onError:false,message:''}) 
    const [ready,SetReady] = useState(false)

    const handleCloseAlert = (event, reason) => {
        if (reason === 'clickaway') {
          return;
        }
        SetReady(false);
        SetError({onError:false,message:''});
      };
    
    
    const Styles = {

        LoginPaper:{
            margin:"0 auto",
            marginTop:"40px",
            marginBottom:"40px",
            maxWidth:"420px",
            height:"320px"
        },

        LoginForm:{
            padding:"20px"
        },
        
        InputItem : {
            margin:"10px" 
        },
    }

    const handleSubmit = async (e) =>{

        e.preventDefault()
        const res = await LOGIN(email,password)
            //console.log(res)
            if(res.data.data.login!==null){
                
                SetReady(true)
            }
            else if(res.data.data.errors!==null){
                const err = res.data.errors
                err.map(item => {
                    return SetError({onError:true,message:item.message})
                    
                })
            }
    }

    return (
        <div>
        <Paper style={Styles.LoginPaper}>
            {props.handleSubmit ?props.handleSubmit():""}
            <CardHeader title="Login" className="addgame-card-header"/>

            <form onSubmit={handleSubmit}
            style={Styles.LoginForm}>
                <input 
                id="email"
                type="text" 
                name="email"
                value={props.email}
                style={Styles.InputItem}  
                onChange={e=>SetEmail(e.target.value)} />

                <input 
                id="password"
                type="password" 
                name="password"
                value={props.password}
                style={Styles.InputItem} 
                onChange={e=>SetPassword(e.target.value)}/>

                <Button type="submit" id="submit"
                variant="contained" color="primary" style={Styles.InputItem} >
                    Enter
                </Button>

            {error.onError ?
            <Alert onClose={handleCloseAlert} severity="error">
              {error.message}
            </Alert>

          :""}

            </form>

            {ready === true ? 
            <Redirect to="/processlogin"/>
            :""}


        </Paper>
        
        <Footer/>

        </div>
    
    
    )
}

export default Login