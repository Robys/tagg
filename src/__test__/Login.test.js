import React from "react";
import Login from '../views/Login'
import {shallow} from '../enzyme'


describe('rendering <Login/>', () =>{
    
    const email = "adm@tagg.com"
    const password = "123456"

    const container = shallow(<Login email={email} password={password} />)
    const handleSubmit = jest.fn(() => console.log("mock called"));

    test('rendering HTML without crashing',()=>{
        expect(container).toMatchSnapshot()
    })
        
    it('testing email input',()=>{
        //find email input in the login component and simulate typing
        const input = container.find('input[type="text"]')
        //compare input with expected value
        expect(input.props().value).toEqual(email)
    })

    it('testing password input',()=>{
        //find password input in the login component and simulate typing
        const input = container.find('input[type="password"]')
        //compare input with expected value
        expect(input.props().value).toEqual(password)
    })

    it('test submit function', ()=>{
        const wrapper = shallow(<Login handleSubmit={handleSubmit} />)
        wrapper.find('#submit').simulate('click')
        wrapper.update()
        expect(handleSubmit).toBeCalled();
    })

})


