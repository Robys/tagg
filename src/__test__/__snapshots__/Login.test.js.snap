// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering <Login/> rendering HTML without crashing 1`] = `
<WithStyles(ForwardRef(Paper))
  style={
    Object {
      "height": "320px",
      "margin": "0 auto",
      "marginTop": "80px",
      "maxWidth": "420px",
    }
  }
>
  <WithStyles(ForwardRef(CardHeader))
    className="addgame-card-header"
    title="Login"
  />
  <form
    onSubmit={[Function]}
    style={
      Object {
        "padding": "20px",
      }
    }
  >
    <input
      id="email"
      name="email"
      onChange={[Function]}
      style={
        Object {
          "margin": "10px",
        }
      }
      type="text"
      value="adm@tagg.com"
    />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      style={
        Object {
          "margin": "10px",
        }
      }
      type="password"
      value="123456"
    />
    <WithStyles(ForwardRef(Button))
      color="primary"
      id="submit"
      style={
        Object {
          "margin": "10px",
        }
      }
      type="submit"
      variant="contained"
    >
      Enter
    </WithStyles(ForwardRef(Button))>
  </form>
</WithStyles(ForwardRef(Paper))>
`;
